### PR TITLE
Add Inference Dataset to Training

### PR DESCRIPTION
**Pull request by Sean1572**
Adds a secondary validation dataset to training process in order to estimate domain shift from xento-canto to soundscapes

### CODE REVIEW

File: pyha_analyzer/dataset.py [hunk 1]
```diff
@@ -427,9 +431,16 @@ def main() -> None:
     """
     testing function.
     """
+    run = wandb.init(
+            entity=cfg.wandb_entity,
+            project=cfg.wandb_project,
+            config=cfg.config_dict,
+            mode="online" if cfg.logging else "disabled")
+    run.name = "inferance testing"
```
Comment (benjamin-cates) [inline pyha_analyzer/dataset.py]: Ooop remove this line, should be in csv

File: .pylintrc [hunk 2]
```diff
@@ -278,7 +278,7 @@ exclude-too-few-public-methods=
 ignored-parents=
 
 # Maximum number of arguments for function / method.
-max-args=5
+max-args=7
```
Comment (benjamin-cates) [inline .pylintrc:281]: BASED

Comment (Sean1572) [reply .pylintrc:281]: This doesn't help us, and increasing doesn't really hurt us

**Comment by benjamin-cates** [comment]
Whoever merges this PLEASE squash it 😭

File: pyha_analyzer/dataset.py [hunk 3]
```diff
@@ -349,15 +360,26 @@ def get_datasets():
     train_ds = PyhaDFDataset(train, train=True, species=classes)
 
     valid_ds = PyhaDFDataset(valid, train=False, species=classes)
-    return train_ds, valid_ds
+
+
+
+    #Handle inferance datasets
```
Comment (marco-bazzani) [inline pyha_analyzer/dataset.py]: Inference

File: pyha_analyzer/train.py [hunk 4]
```diff
@@ -217,8 +219,52 @@ def valid(model: Any,
         logger.info("Model saved in: %s", save_model(model))
         logger.info("Validation mAP Improved - %f ---> %f", BEST_VALID_MAP, valid_map)
         BEST_VALID_MAP = valid_map
+
+
+    inferance_valid(model, infer_loader, epoch_progress, valid_map)
```
Comment (marco-bazzani) [inline pyha_analyzer/train.py]: inference

File: pyha_analyzer/train.py [hunk 5]
```diff
@@ -217,8 +219,52 @@ def valid(model: Any,
...
     return valid_map
 
+
+def inferance_valid(model: Any,
```
Comment (marco-bazzani) [inline pyha_analyzer/train.py]: inference

**Review by marco-bazzani — CHANGES_REQUESTED** [review]

**Comment by Sean1572** [comment]
> Whoever merges this PLEASE squash it 😭

Gotcha, will not squash it

**Review by marco-bazzani — APPROVED** [review]

**Comment by marco-bazzani** [comment]
Wait until testing is finished

**Pull request merged**